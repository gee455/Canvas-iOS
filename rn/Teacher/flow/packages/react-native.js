//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
