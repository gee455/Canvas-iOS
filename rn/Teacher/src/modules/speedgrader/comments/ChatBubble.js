//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  Image,
} from 'react-native'
import {
  Text,
} from '../../../common/text'
import images from '../../../images'
import colors from '../../../common/colors'

type BubbleProps = {
  message: string,
  from: 'me' | 'them',
}

export default class ChatBubble extends Component<BubbleProps, any> {
  render () {
    let image
    let fromStyle
    let messageContainer
    if (this.props.from === 'me') {
      image = images.speedGrader.chatBubbleMe
      fromStyle = styles.myText
      messageContainer = styles.myMessageContainer
    } else {
      image = images.speedGrader.chatBubbleThem
      fromStyle = styles.theirText
      messageContainer = styles.theirMessageContainer
    }

    return (
      <View style={messageContainer}>
        <Image
          source={image}
          style={styles.bubble}
          capInsets={{ left: 18, right: 18, top: 24, bottom: 16 }}
          resizeMode='stretch'
        />
        <Text style={[styles.message, fromStyle]}>
          {this.props.message}
        </Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  myText: {
    color: 'white',
  },
  theirText: {
    color: colors.darkText,
  },
  myMessageContainer: {
    flex: 1,
    maxWidth: 300,
  },
  theirMessageContainer: {
    flex: 1,
    maxWidth: 300,
  },
  message: {
    paddingHorizontal: 12,
    paddingTop: 12,
    paddingBottom: 8,
    backgroundColor: 'transparent',
  },
  bubble: {
    position: 'absolute',
    margin: 0,
    padding: 0,
    top: -4,
    bottom: 0,
    left: 0,
    right: 0,
    width: null,
    height: null,
  },
})
