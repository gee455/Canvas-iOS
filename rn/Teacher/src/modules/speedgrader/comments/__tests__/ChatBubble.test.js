//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import ChatBubble from '../ChatBubble'

test('my chat bubbles render correctly', () => {
  let tree = renderer.create(
    <ChatBubble from="me" message="Hello, World!"/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('their chat bubbles render correctly', () => {
  let tree = renderer.create(
    <ChatBubble from="them" message="Hello, back!" />
  )
  expect(tree).toMatchSnapshot()
})
