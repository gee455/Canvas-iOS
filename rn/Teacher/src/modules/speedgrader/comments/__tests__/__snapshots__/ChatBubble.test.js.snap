// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my chat bubbles render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "maxWidth": 300,
    }
  }
>
  <Image
    capInsets={
      Object {
        "bottom": 16,
        "left": 18,
        "right": 18,
        "top": 24,
      }
    }
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../src/images/speedgrader/bubble-me.png",
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": null,
        "left": 0,
        "margin": 0,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": -4,
        "width": null,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "paddingBottom": 8,
            "paddingHorizontal": 12,
            "paddingTop": 12,
          },
          Object {
            "color": "white",
          },
        ],
        Object {
          "fontFamily": ".SFUIDisplay",
        },
      ]
    }
  >
    Hello, World!
  </Text>
</View>
`;

exports[`their chat bubbles render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "maxWidth": 300,
    }
  }
>
  <Image
    capInsets={
      Object {
        "bottom": 16,
        "left": 18,
        "right": 18,
        "top": 24,
      }
    }
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../src/images/speedgrader/bubble-them.png",
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": null,
        "left": 0,
        "margin": 0,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": -4,
        "width": null,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": ".SFUIDisplay",
        },
        Object {
          "color": "#2D3B45",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "paddingBottom": 8,
            "paddingHorizontal": 12,
            "paddingTop": 12,
          },
          Object {
            "color": "#2D3B45",
          },
        ],
        Object {
          "fontFamily": ".SFUIDisplay",
        },
      ]
    }
  >
    Hello, back!
  </Text>
</View>
`;
